core/render_management.ts

```diff
@@ -43,11 +44,22 @@ function queueBlock(block: BlockSvg) {
  * Rerenders all of the blocks in the queue.
  */
 function doRenders() {
+  const workspaces = new Set([...rootBlocks].map((block) => block.workspace));
   for (const block of rootBlocks) {
+    // No need to render a dead block.
     if (block.isDisposed()) continue;
+    // Rendering will be triggered through the real root block.
```

Comment:
Like we discussed offline, consider adding a comment explaining why a "root" block isn't a "real root block"

Reply:
Tell me if this is better!